src/models.ts

```diff
@@ -198,7 +198,7 @@ export abstract class Filter {
     this.target = target;
   }
 
-  toJSON(): IFilter {
+  toJSON(): any {
```

Comment:
Thoughts on this change?  You can't return a type.  Looks like this is only used internally by the implementations of basic filters & advanced filter so should be OK.